### PR TITLE
Fix .bazelrc logic when bazel_workspace_dir is set

### PR DESCRIPTION
**Pull request by bduffany**
`buildbuddy.yaml` supports a `bazel_workspace_dir` option for testing repos that contain more than one bazel workspace (e.g. rules_xcodeproj contains multiple workspaces to run various integration tests).

We weren't taking this into account when computing the bazel args used to apply our custom bazelrc, meaning we'd treat the workspace bazelrc with lower precedence than we should have.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/cmd/ci_runner/main.go [hunk 1]
```diff
@@ -717,9 +717,13 @@ func run() error {
 		if err != nil {
 			log.Warningf("Failed to read BuildBuddy config; will run `bazel shutdown` from repo root: %s", err)
 		}
-		wsPath := ""
+		cwd, err := os.Getwd()
```
Comment (maggie-lou) [inline enterprise/server/cmd/ci_runner/main.go:720]: We do a lot of changing directories, so this logic feels a bit brittle. Could we reuse this logic here `bazelWorkspaceAbsPath := filepath.Join(ws.rootDir, repoDirName, action.BazelWorkspaceDir)` and consolidate it in `bazelArgsWithCustomBazelrc`?

File: enterprise/server/cmd/ci_runner/main.go [hunk 2]
```diff
@@ -2521,8 +2521,13 @@ func runBazelWrapper() error {
 		return err
 	}
 
+	// Get the current bazel workspace path where we expect to find the
+	// workspace rc file. If this fails just treat it like we aren't in a
+	// workspace.
+	workspacePath, _ := currentBazelWorkspaceAbsPath()
```
Comment (maggie-lou) [inline enterprise/server/cmd/ci_runner/main.go:2527]: Should we at least log the error to the executor?

File: enterprise/server/cmd/ci_runner/main.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/cmd/ci_runner/main.go]: Done

File: enterprise/server/cmd/ci_runner/main.go (hunk 2 above)
Comment (bduffany) [reply enterprise/server/cmd/ci_runner/main.go]: Changed it to just return an error - if there's an error here other than NotFound then it's probably some sort of filesystem error, and it shouldn't matter much whether we return it here or if we let bazel run and return the error

**Review by maggie-lou — APPROVED** [review]

**Pull request merged**